### ISSUE TITLE
Documentation for CSS Features

### ISSUE THREAD

**Issue by onigoetz** [swc-project/swc#3900]
### Describe the feature

I saw that the changelog has many mentions for CSS features (parser, linter, minifier )

I would be very interested to see the documentation but couldn't find it, is there anything published yet ?

And some side questions
* There is the [@parcel/css](https://www.npmjs.com/package/@parcel/css) package that seems to provide similar CSS features, is there some collaboration between the two projects ?
* Does it have to be in the same bundle ? (related to #3899) I would be very interested to have the CSS linter for example in a project but wouldn't need the JavaScript compiler (just an example)

**Comment by kdy1** [comment]
It's not published yet

**Comment by onigoetz** [comment]
It's not only a bundler and minifier. They also provide vendor prefixing and fallbacks (nesting, css colors level 4 fallback and more) 
I'm also pretty sure they want to add plugin support.

@devongovett built it. He certainly knows better what will and will not come in `@parcel/css` future

**Comment by devongovett** [comment]
Yeah, Parcel CSS already has minification, bundling, auto prefixing, and syntax lowering (like preset-env). We expose a Rust [crate](https://docs.rs/parcel_css/1.0.0-alpha.18/parcel_css/index.html) which gives you full access to parsed structures, and you can already manipulate them to implement custom transforms. We'll be improving the API in the future to support [custom syntax extensions](https://github.com/parcel-bundler/parcel-css/pull/94) etc., and offering a JS API as well.

A main difference with SWC how parsing is done. Parcel CSS works more like browsers and fully parses all known properties, whereas SWC only provides a more generic AST that each consumer must interpret themselves. I discussed the differences in approach that Parcel CSS takes vs other tools in our announcement blog post: https://parceljs.org/blog/parcel-css/#architecture.

I suggested working together on it in the past (#2884), but the SWC developers wanted to go in a different direction.

**Comment by alexander-akait** [comment]
@devongovett Not fully truth, we use the both approaches, trying to parser known and provide more better AST structures where we known how to parser, if we can't parser we provide generic AST (at the moment we are not processing too many things, but it designed to do it and we are continuing improve it).

We want to help not only bundling, our area is more - linter/formatter/etc so we need AST here and some more generic structures (especial for recovery mode), so author can write plugins for future CSS syntax without problems.

> Parcel CSS works more like browsers and fully parses all known properties, whereas SWC provides a more generic AST that each consumer must interpret themselves. 

This statement is not entirely correct. We also parser like a browser, just what the browser cannot parse, we put it in common AST structures, as I said earlier.

> I suggested working together on it in the past (https://github.com/swc-project/swc/pull/2884), but the SWC developers wanted to go in a different direction.

No one refused to help, you want to develop it in another repository, like a side project, unfortunately, this approach in organizational terms does not suit us
